wagtail/core/models.py

```diff
@@ -115,6 +120,8 @@ def _copy(source, exclude_fields=None, update_attrs=None):
 
     if update_attrs:
         for field, value in update_attrs.items():
+            if field not in data_dict:
+                continue
```

Comment:
I think there's still a possible failure case when `update_attrs` includes a foreign key field referenced as `foo_id`, because `data_dict` only includes the `foo` variant. However, in the case of pages at least, it looks like this fixes itself further down when we create a new revision:
https://github.com/wagtail/wagtail/blob/5e6a6746869d351eb2c55e092d6e669da1af896f/wagtail/core/models.py#L1493-L1495

Given the amount of reaching I'm doing to try (and fail) to turn this into a failing test case, and the unlikeliness of this happening in the real world (I'm pretty sure that `update_attrs` is only going to be used on well-established fields like `slug`), I'm going to call this good enough :-)